src/App/Fossa/Reachability/Jar.hs

```diff
@@ -74,6 +78,20 @@ instance ToDiagnostic FailedToParseJar where
   renderDiagnostic (FailedToParseJar jar) =
     createErrataWithHeaderOnly $ "Could not read from jar, so skipping: " <> toText (show jar)
 
+-- | Like @mavenJarCallGraph@, but used when the list of JARs to parse is already available
+-- and works for any Java project.
+callGraphFromJars ::
+  ( Has Logger sig m
+  , Has Diagnostics sig m
+  , Has Exec sig m
+  , Has (Lift IO) sig m
+  ) =>
+  [Path Abs File] ->
+  m CallGraphAnalysis
+callGraphFromJars jars = context ("build call graph from " <> toText (show jars)) $ do
+  parsedJars <- traverse callGraphFromJar jars
+  pure $ JarAnalysis (catMaybes parsedJars)
```

Comment:
[Prefer] I think we should fail, if we do not find at least on Jar in `parsedJar`. I should have done this in previous pr, but it is possible to have empty `jars` from config now as well. 

Ideally, if you have cycles - we can make `JarAnalysis NonEmpty ...`, but this will increase scope of PR - I can address this in next batch of PR work.

Reply:
🤔 we're trying to fail stuff less often, and just upload partial data when we have it. why would we reverse course on this?

Reply:
It gives wrong impression in scan summary since the `Result Success _ (...JarAnalysis [])`